integrations/SnapEngage/browser.js

```diff
@@ -0,0 +1,202 @@
+      window.rudderanalytics.track(
+        "Live Chat Message Received",
+        { agentUsername: agent },
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("ChatMessageSent", function () {
+      window.rudderanalytics.track(
+        "Live Chat Message Sent",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("Close", function () {
+      window.rudderanalytics.track(
+        "Live Chat Conversation Ended",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+
+    window.SnapEngage.setCallback("InlineButtonClicked", function () {
+      window.rudderanalytics.track(
+        "Inline Button Clicked",
+        {},
+        { context: { integration: integrationContext } }
+      );
+    });
+  }
+
+  recordingLiveChatEventsWithMapping(updateEventNames, standardEventsMap) {
+    each((val, key) => {
+      window.SnapEngage.setCallback("StartChat", function () {
+        let eventName = "Live Chat Conversation Started";
+        if (updateEventNames && val === "startChat") {
+          eventName = key;
+        }
+        window.rudderanalytics.track(
+          `${eventName}`,
+          {},
+          { context: { integration: integrationContext } }
+        );
+      });
+
+      window.SnapEngage.setCallback("ChatMessageReceived", function (agent) {
+        let eventName = "Live Chat Message Received";
+        if (updateEventNames && val === "chatMessageReceived") {
+          eventName = key;
+        }
+        window.rudderanalytics.track(
+          `${eventName}`,
+          { agentUsername: agent },
+          { context: { integration: integrationContext } }
+        );
+      });
+
+      window.SnapEngage.setCallback("ChatMessageSent", function () {
+        let eventName = "Live Chat Message Sent";
+        if (updateEventNames && val === "chatMessageSent") {
+          eventName = key;
+        }
+        window.rudderanalytics.track(
+          `${eventName}`,
+          {},
+          { context: { integration: integrationContext } }
+        );
+      });
+
+      window.SnapEngage.setCallback("Close", function () {
+        let eventName = "Live Chat Conversation Ended";
+        if (updateEventNames && val === "close") {
+          eventName = key;
+        }
+        window.rudderanalytics.track(
+          `${eventName}`,
+          {},
+          { context: { integration: integrationContext } }
+        );
+      });
+
+      window.SnapEngage.setCallback("InlineButtonClicked", function () {
+        let eventName = "Inline Button Clicked";
+        if (updateEventNames && val === "inlineButtonClicked") {
+          eventName = key;
+        }
+        window.rudderanalytics.track(
+          `${eventName}`,
+          {},
+          { context: { integration: integrationContext } }
+        );
+      });
+    }, standardEventsMap);
+  }
+}
```

Comment:
This part is repeating try to combine it with `recordingLiveChatEvents` function.

Reply:
Done